extensions/rich_text_components/Skillreview/directives/oppia-noninteractive-skillreview.directive.ts

```diff
@@ -34,6 +34,19 @@ angular.module('oppia').directive('oppiaNoninteractiveSkillreview', [
       restrict: 'E',
       scope: {},
       bindToController: {},
+      link: function(scope, elem, attrs) {
+        // In the editor, this directive is enclosed in a span tag which doesn't
+        // add a whitespace after the element, but when previewing the element,
+        // it is enclosed in a <p> tag, which does add a whitespace, hence the
+        // check.
+        if (
+          elem.find(
+            '.skillreview-rte-element'
+          ).parent().parent()[0].localName === 'p'
+        ) {
+          elem.find('.skillreview-rte-element').css('margin-right', '-4px');
```

Comment:
This is a bit hacky. I'm fine with leaving it in for now because it fixes an important bug, but probably we should add a TODO because I think this is going to be problematic with the angular migration, particularly the reference to parent(). Not sure though -- /cc @srijanreddy98. 

One question though -- wouldn't it be simpler to always add a margin-right CSS (with explanation) directly in the html?

Reply:
Done (as per discussion). Removed this and editing this via CSS now.